### PR TITLE
Support for rigid couplings

### PR DESCRIPTION
**Pull request by kmokstad**
This adds the support for introducing a rigid coupling from a patch boundary to a spatial point, which then can be considered a nodal point with 6 degrees of freedom (including rotations). Handy for modeling different end support for structures modeled with continuum elements. Needed for the OO bridge and perhaps also for the reservoir drill hole with casing inserted.

### CODE REVIEW

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=108 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=108 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=108 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=108 please

File: src/SIM/SIMinput.C [hunk 1]
```diff
@@ -272,6 +272,17 @@ bool SIMinput::parseGeometryTag (const TiXmlElement* elem)
               }
             }
         }
+
+        item = set->FirstChildElement("point");
+        for (; idim == 0 && item; item = item->NextSiblingElement("point"))
+          if (item->FirstChild())
```
Comment (akva2) [inline src/SIM/SIMinput.C:281]: if somebody for some reason have a point tag with no value, this will crash.

Comment (kmokstad) [reply src/SIM/SIMinput.C]: That's not supposed to happen, but put in check on `item->FirstChild()->Value()`anyway

File: src/ASM/ASMbase.h [hunk 2]
```diff
@@ -874,8 +893,9 @@ class ASMbase
   static int gNod; //!< Global node counter
 
 private:
-  std::pair<size_t,size_t> myLMs; //!< Nodal range of the Lagrange multipliers
-  std::vector<char>    myLMTypes; //!< Type of Lagrange multiplier ('L' or 'G')
+  std::vector<char> myLMTypes; //!< Type of Lagrange multiplier ('L' or 'G')
+  std::set<size_t>  myLMs;     //!< Nodal indices of the Lagrange multipliers
```
Comment (akva2) [inline src/ASM/ASMbase.h:897]: are you sure we are not order dependent anywhere?

Comment (kmokstad) [reply src/ASM/ASMbase.h:897]: Not sure what you mean

File: src/ASM/ASMs3D.h [hunk 3]
```diff
@@ -370,7 +371,7 @@ class ASMs3D : public ASMstruct, public ASM3D
   virtual void closeBoundaries(int dir, int basis, int master);
 
   //! \brief Collapses a degenereated face into a single node or edge.
```
Comment (akva2) [inline src/ASM/ASMs3D.h:373]: kan vel like godt fikse typo her også da når du er i gang.

Comment (kmokstad) [reply src/ASM/ASMs3D.h]: Sure. Har tydligvis ikke kjørt spell-check på kommentarlinjer. Fixed.

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=108 ifem-finitedeformation=25 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=108 ifem-finitedeformation=25 please

**Pull request merged**